Makefile

```diff
@@ -8,4 +8,39 @@ ifneq (,$(shell grep "path *= *$(LIBDIR)" .gitmodules 2>/dev/null))
 else
 	git clone -q --depth 10 $(CLONE_ARGS) \
 	    -b main https://github.com/martinthomson/i-d-template $(LIBDIR)
-endif
\ No newline at end of file
+endif
+
+# run cddl_validate.sh for all drafts_source files 
+cddl:
+	@for f in $(drafts_source); do \
+	    echo "Validating $$f"; \
+	    ./cddl_validate.sh $$f > /tmp/foo 2>&1 ; \
+	    if [ $$? -eq 0 ]; then \
+	        echo "  OK"; \
+	    else \
+	        echo "  ERROR"; \
+					echo "  debug with: ./cddl_validate.sh $$f"; \
+	    fi; \
+	done
+
+# override lib/main.mk deps target, to also install deps from Gemfile
+.PHONY: deps
+deps::
+		$(MAKE) -f $(LIBDIR)/main.mk $@
+		bundle install --gemfile=$(realpath $<)
```

Comment:
very good point. 
I think checking and throwing an error is the best course of action. 

Installing ruby or python is something that the lib makefile also doesn't do because it would get into the multi-platform rabbit hole. 
I'll cqll it out of scope 😅 and just check for ruby

Reply:
turns out the lib makefile is pure magic and it already does everything we want just by already checking for the presence of a local Gemfile .. MT for the win.